pkg/controllers/nodeclaim/termination/suite_test.go

```diff
@@ -216,4 +217,30 @@ var _ = Describe("Termination", func() {
 			ExpectExists(ctx, env.Client, node)
 		}
 	})
+	It("should retry nodeClaim deletion when cloudProvider returns retryable error", func() {
+		ExpectApplied(ctx, env.Client, nodePool, nodeClaim)
+		ExpectReconcileSucceeded(ctx, nodeClaimLifecycleController, client.ObjectKeyFromObject(nodeClaim))
+
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		_, err := cloudProvider.Get(ctx, nodeClaim.Status.ProviderID)
+		Expect(err).ToNot(HaveOccurred())
+
+		node := test.NodeClaimLinkedNode(nodeClaim)
+		ExpectApplied(ctx, env.Client, node)
+
+		// Expect the node and the nodeClaim to both be gone
+		Expect(env.Client.Delete(ctx, nodeClaim)).To(Succeed())
+		ExpectReconcileSucceeded(ctx, nodeClaimTerminationController, client.ObjectKeyFromObject(nodeClaim)) // triggers the node deletion
+		ExpectFinalizersRemoved(ctx, env.Client, node)
+		ExpectNotFound(ctx, env.Client, node)
+
+		cloudProvider.NextDeleteErr = cloudprovider.NewRetryableError(fmt.Errorf("underlying instance not terminated"))
+		ExpectReconcileFailed(ctx, nodeClaimTerminationController, client.ObjectKeyFromObject(nodeClaim)) // re-enqueue reconciliation due to retryable error
```

Comment:
This is good test; however, we should think about what our back-off strategy is going to be here and ensure that we are returning the correct result here with this change.